infrastructure/sandbox/PreProvisioner/lambda/main.go

```diff
@@ -8,23 +8,82 @@ import (
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/dynamodb"
 	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/fleetdm/fleet/v4/orbit/pkg/packaging"
+	"github.com/fleetdm/fleet/v4/server"
+	"github.com/fleetdm/fleet/v4/server/config"
+	"github.com/fleetdm/fleet/v4/server/datastore/s3"
+	"github.com/fleetdm/fleet/v4/server/fleet"
 	"github.com/google/uuid"
 	flags "github.com/jessevdk/go-flags"
 	"log"
 	"math/rand"
 	"os"
 	"os/exec"
+	"path/filepath"
 )
 
 type OptionsStruct struct {
 	LambdaExecutionEnv string `long:"lambda-execution-environment" env:"AWS_EXECUTION_ENV"`
 	LifecycleTable     string `long:"dynamodb-lifecycle-table" env:"DYNAMODB_LIFECYCLE_TABLE" required:"true"`
 	MaxInstances       int64  `long:"max-instances" env:"MAX_INSTANCES" required:"true"`
 	QueuedInstances    int64  `long:"queued-instances" env:"QUEUED_INSTANCES" required:"true"`
+	FleetBaseURL       string `long:"fleet-base-url" env:"FLEET_BASE_URL" required:"true"`
+	InstallerBucket    string `long:"installer-bucket" env:"INSTALLER_BUCKET" required:"true"`
 }
 
 var options = OptionsStruct{}
 
+func buildPackages(instanceID, enrollSecret string) (err error) {
+	funcs := []func(packaging.Options) (string, error){
+		packaging.BuildPkg,
+		packaging.BuildDeb,
+		packaging.BuildRPM,
+		packaging.BuildMSI,
+	}
+	pkgopts := packaging.Options{
+		FleetURL:     fmt.Sprintf("https://%s.%s", options.FleetBaseURL, instanceID),
+		EnrollSecret: enrollSecret,
+	}
+	store, err := s3.NewInstallerStore(config.S3Config{
+		Bucket: options.InstallerBucket,
+	})
```

Comment:
it's super cool that you're able to use this directly! A heads-up that you might be already aware of: the config flags you provide for this (eg: `FLEET_PACKAGING_S3_PREFIX`) are used by the Fleet server when it initializes an `InstallerStore` instance

https://github.com/fleetdm/fleet/blob/db85569791605c061284b1533cf96f319d131144/cmd/fleet/serve.go#L211-L215

Since you're using this as a library, you'll have to provide the same config values here (probably the most critical is the s3 prefix), all valid options are https://github.com/fleetdm/fleet/blob/db85569791605c061284b1533cf96f319d131144/server/config/config.go#L229-L239

Reply:
Yeah, I copied from the tools one. Since this is running inside of amazon we shouldn't need to provide any config except for maybe prefix as it can auto gather that info.

Reply:
Gotcha, just be aware that (and it can be changed if it's a burden) the `InstallerStore` instance builds the key using the prefix provided on initialization:

https://github.com/fleetdm/fleet/blob/db85569791605c061284b1533cf96f319d131144/server/datastore/s3/installer.go#L78-L85